pycbc/inference/sampler/dynesty.py

```diff
@@ -0,0 +1,232 @@
+    package.
+
+    Parameters
+    ----------
+    model : model
+        A model from ``pycbc.inference.models``.
+    nlive : int
+        Number of live points to use in sampler.
+    dlogz: float
+        Tolerance limit to the value of logz (also a convergence criteria)
+    pool : function with map, Optional
+        A provider of a map function that allows a function call to be run
+        over multiple sets of arguments and possibly maps them to
+        cores/nodes/etc.
+    """
+    name = "dynesty"
+    _io = DynestyFile
+
+    def __init__(self, model, nlive, err_logz, nprocesses=1,
+                 loglikelihood_function=None, use_mpi=False, **kwargs):
+
+        self.model = model 
+        # Set up the pool
+        model_call = DynestyModel(model, loglikelihood_function)
+        if nprocesses > 1:
+            # these are used to help paralleize over multiple cores / MPI
+            models._global_instance = model_call
+            log_likelihood_call = _call_global_loglikelihood
+            prior_call = _call_global_logprior
+        else:
+            prior_call = model_call.prior_transform
+            log_likelihood_call = model_call.log_likelihood
+        pool = choose_pool(mpi=use_mpi, processes=nprocesses)
+        if pool is not None:
+            pool.size = nprocesses
+
+        self.nlive = nlive
+        self.err_logz = err_logz
+        self.names = model.sampling_params
+        self.ndim = len(model.sampling_params)
+        self.checkpoint_file = None
+        self._sampleri = dynesty.NestedSampler(log_likelihood_call,
+                                               prior_call, self.ndim,
+                                               nlive=self.nlive,
+                                               dlogz=self.err_logz,
+                                               pool=pool, **kwargs)
+
+    def run(self):
+        res = self._sampler.run_nested()
+
+    @property
+    def io(self):
+        return self._io
+
+    @property
+    def niterations(self):
+        return len(tuple(self.samples.values())[0])
+
+    @classmethod
+    def from_config(cls, cp, model, nprocesses=1, loglikelihood_function=None,
+                    use_mpi=False):
+        """
+        Loads the sampler from the given config file.
+        """
+        section = "sampler"
+        # check name
+        assert cp.get(section, "name") == cls.name, (
+            "name in section [sampler] must match mine")
+        # get the number of live points to use
+        nlive = int(cp.get(section, "nlive"))
+        err_logz = float(cp.get(section, "err_logz"))
+        loglikelihood_function = \
+            get_optional_arg_from_config(cp, section, 'loglikelihood-function')
+        obj = cls(model, nlive=nlive, err_logz=err_logz, nprocesses=nprocesses,
+                  loglikelihood_function=loglikelihood_function,
+                  use_mpi=use_mpi)
+        return obj
+
+    def checkpoint(self):
+        pass
```

Comment:
Does dynesty support checkpointing?

Reply:
It support checkpointing but I will open another pull request to work on checkpointing after I am done with adding support for dynasty sampler in its most basic form.